src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -248,22 +259,32 @@ def acs_create(resource_group_name, deployment_name, dns_name_prefix, name, ssh_
     :raises: :class:`CloudError<msrestazure.azure_exceptions.CloudError>`
     """
     if orchestrator_type == 'Kubernetes' or orchestrator_type == 'kubernetes':
-        principalObj = load_acs_service_principal()
-        if principalObj:
-            service_principal = principalObj.get('service_principal')
-            client_secret = principalObj.get('client_secret')
-
+        # TODO: This really needs to be broken out and unit tested.
         if not service_principal:
+            # --service-principal not specified, try to load it from local disk
+            principalObj = load_acs_service_principal()
+            if principalObj:
+                service_principal = principalObj.get('service_principal')
+                client_secret = principalObj.get('client_secret')
+                _validate_service_principal(service_principal)
+            else:
+                # Nothing to load, make one.
+                if not client_secret:
+                    client_secret = binascii.b2a_hex(os.urandom(10)).decode('utf-8')
+                store_acs_service_principal(client_secret, None)
```

Comment:
What's the purpose of calling `store_acs_service_principal` twice (lines 274, 280)?